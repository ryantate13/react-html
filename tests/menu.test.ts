import {menu} from '../src';

describe('menu', () => {
    it('renders with children', () => {
        expect(menu({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(menu({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(menu({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(menu()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(menu({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(menu({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(menu({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(menu({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(menu({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(menu({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(menu({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(menu({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(menu({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(menu({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(menu({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(menu({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(menu({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(menu({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(menu({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(menu({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(menu({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(menu({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(menu({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(menu({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(menu({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(menu({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(menu({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(menu({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(menu({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(menu({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(menu({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(menu({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(menu({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(menu({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(menu({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(menu({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(menu({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(menu({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(menu({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(menu({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(menu({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(menu({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(menu({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(menu({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(menu({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(menu({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(menu({translate: "no"})).toMatchSnapshot();
    });
});