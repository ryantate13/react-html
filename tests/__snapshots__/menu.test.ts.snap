// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu renders with accessKey="a,b,c,d" 1`] = `
<menu
  accessKey="a,b,c,d"
/>
`;

exports[`menu renders with autoCapitalize="characters" 1`] = `
<menu
  autoCapitalize="characters"
/>
`;

exports[`menu renders with autoCapitalize="none" 1`] = `
<menu
  autoCapitalize="none"
/>
`;

exports[`menu renders with autoCapitalize="off" 1`] = `
<menu
  autoCapitalize="off"
/>
`;

exports[`menu renders with autoCapitalize="on" 1`] = `
<menu
  autoCapitalize="on"
/>
`;

exports[`menu renders with autoCapitalize="sentences" 1`] = `
<menu
  autoCapitalize="sentences"
/>
`;

exports[`menu renders with autoCapitalize="words" 1`] = `
<menu
  autoCapitalize="words"
/>
`;

exports[`menu renders with children 1`] = `
<menu>
  TEST
</menu>
`;

exports[`menu renders with className="foo bar baz" 1`] = `
<menu
  className="foo bar baz"
/>
`;

exports[`menu renders with contentEditable="false" 1`] = `
<menu
  contentEditable="false"
/>
`;

exports[`menu renders with contentEditable="true" 1`] = `
<menu
  contentEditable="true"
/>
`;

exports[`menu renders with dir="auto" 1`] = `
<menu
  dir="auto"
/>
`;

exports[`menu renders with dir="ltr" 1`] = `
<menu
  dir="ltr"
/>
`;

exports[`menu renders with dir="rtl" 1`] = `
<menu
  dir="rtl"
/>
`;

exports[`menu renders with draggable="false" 1`] = `
<menu
  draggable="false"
/>
`;

exports[`menu renders with draggable="true" 1`] = `
<menu
  draggable="true"
/>
`;

exports[`menu renders with empty attributes 1`] = `<menu />`;

exports[`menu renders with hidden="false" 1`] = `
<menu
  hidden={false}
/>
`;

exports[`menu renders with hidden="true" 1`] = `
<menu
  hidden={true}
/>
`;

exports[`menu renders with id="TEST" 1`] = `
<menu
  id="TEST"
/>
`;

exports[`menu renders with inputMode="decimal" 1`] = `
<menu
  inputMode="decimal"
/>
`;

exports[`menu renders with inputMode="email" 1`] = `
<menu
  inputMode="email"
/>
`;

exports[`menu renders with inputMode="none" 1`] = `
<menu
  inputMode="none"
/>
`;

exports[`menu renders with inputMode="numeric" 1`] = `
<menu
  inputMode="numeric"
/>
`;

exports[`menu renders with inputMode="search" 1`] = `
<menu
  inputMode="search"
/>
`;

exports[`menu renders with inputMode="tel" 1`] = `
<menu
  inputMode="tel"
/>
`;

exports[`menu renders with inputMode="text" 1`] = `
<menu
  inputMode="text"
/>
`;

exports[`menu renders with inputMode="url" 1`] = `
<menu
  inputMode="url"
/>
`;

exports[`menu renders with is="foo" 1`] = `
<menu
  is="foo"
/>
`;

exports[`menu renders with itemID="https://google.com/" 1`] = `
<menu
  itemID="https://google.com/"
/>
`;

exports[`menu renders with itemProp="https://google.com/" 1`] = `
<menu
  itemProp="https://google.com/"
/>
`;

exports[`menu renders with itemRef="foo bar" 1`] = `
<menu
  itemRef="foo bar"
/>
`;

exports[`menu renders with itemScope="false" 1`] = `
<menu
  itemScope={false}
/>
`;

exports[`menu renders with itemScope="true" 1`] = `
<menu
  itemScope={true}
/>
`;

exports[`menu renders with itemType="https://google.com/" 1`] = `
<menu
  itemType="https://google.com/"
/>
`;

exports[`menu renders with lang="" 1`] = `
<menu
  lang=""
/>
`;

exports[`menu renders with lang="es-419" 1`] = `
<menu
  lang="es-419"
/>
`;

exports[`menu renders with slot="TEST" 1`] = `
<menu
  slot="TEST"
/>
`;

exports[`menu renders with spellCheck="false" 1`] = `
<menu
  spellCheck="false"
/>
`;

exports[`menu renders with spellCheck="true" 1`] = `
<menu
  spellCheck="true"
/>
`;

exports[`menu renders with style="[object Object]" 1`] = `
<menu
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`menu renders with tabIndex="1" 1`] = `
<menu
  tabIndex={1}
/>
`;

exports[`menu renders with title="TEST" 1`] = `
<menu
  title="TEST"
/>
`;

exports[`menu renders with translate="no" 1`] = `
<menu
  translate="no"
/>
`;

exports[`menu renders with translate="yes" 1`] = `
<menu
  translate="yes"
/>
`;

exports[`menu renders without attributes 1`] = `<menu />`;
